source/common/protobuf/utility.cc

```diff
@@ -656,6 +656,16 @@ void validateDuration(const ProtobufWkt::Duration& duration, int64_t max_seconds
   }
 }
 
+bool validateDurationErrorNoThrow(const ProtobufWkt::Duration& duration,
```

Comment:
nit: consider returning absl::Status and consider making `validateDuration()` call the NoThrow variant and wrapping a validation failure with an exception.

Reply:
Good idea! done